cpp/proto/serializer_base.cc

```diff
@@ -0,0 +1,134 @@
+/* -*- indent-tabs-mode: nil -*- */
+#include "proto/serializer_base.h"
```

Comment:
I think this header should have a more representative name... How about `tls_write.h`, or `tls_encoding.h` (giving a logical place for read functions to get added later, _if necessary_) maybe?